Sources/TuistCoreTesting/MetadataProviders/MockLibraryMetadataProvider.swift

```diff
@@ -3,12 +3,37 @@ import TSCBasic
 @testable import TuistCore
 
 public final class MockLibraryMetadataProvider: MockPrecompiledMetadataProvider, LibraryMetadataProviding {
-    public var productStub: ((LibraryNode) throws -> Product)?
-    public func product(library: LibraryNode) throws -> Product {
-        if let productStub = productStub {
-            return try productStub(library)
+    public var loadMetadataStub: ((AbsolutePath, AbsolutePath, AbsolutePath?) throws -> LibraryMetadata)?
+    public func loadMetadata(at path: AbsolutePath,
+                             publicHeaders: AbsolutePath,
+                             swiftModuleMap: AbsolutePath?) throws -> LibraryMetadata
+    {
+        if let stub = loadMetadataStub {
+            return try stub(path, publicHeaders, swiftModuleMap)
         } else {
-            return .staticLibrary
+            return LibraryMetadata.test(
+                path: path,
+                publicHeaders: publicHeaders,
+                swiftModuleMap: swiftModuleMap
+            )
         }
     }
 }
+
+public extension LibraryMetadata {
```

Comment:
If we end up extracting those into their own files, I'd extract this one into `LibraryMetadata+TestData.swift`.